src/meshes/tree_mesh.jl

```diff
@@ -81,27 +83,29 @@ const ParallelTreeMesh{NDIMS} = TreeMesh{NDIMS, <:ParallelTree{NDIMS}}
 partition!(mesh::SerialTreeMesh) = nothing
 
 # Constructor for passing the dimension and mesh type as an argument
-function TreeMesh(::Type{TreeType},
-                  args...) where {NDIMS, TreeType <: AbstractTree{NDIMS}}
-    TreeMesh{NDIMS, TreeType}(args...)
+function TreeMesh(::Type{TreeType}, args...;
+                  RealT = Float64) where {NDIMS, TreeType <: AbstractTree{NDIMS}}
+    TreeMesh{NDIMS, TreeType, RealT}(args...)
 end
 
 # Constructor accepting a single number as center (as opposed to an array) for 1D
-function TreeMesh{1, TreeType}(n::Int, center::Real, len::Real,
-                               periodicity = true) where {TreeType <: AbstractTree{1}}
-    # TODO: Taal refactor, allow other RealT for the mesh, not just Float64
-    return TreeMesh{1, TreeType}(n, SVector{1, Float64}(center), len, periodicity)
+function TreeMesh{1, TreeType, RealT}(n::Int, center::RealT, len::RealT,
+                                      periodicity = true) where {
+                                                                 TreeType <:
+                                                                 AbstractTree{1},
+                                                                 RealT <: Real}
+    return TreeMesh{1, TreeType, RealT}(n, SVector{1, RealT}(center), len, periodicity)
 end
 
-function TreeMesh{NDIMS, TreeType}(n_cells_max::Integer,
-                                   domain_center::NTuple{NDIMS, Real},
-                                   domain_length::Real,
-                                   periodicity = true) where {NDIMS,
-                                                              TreeType <:
-                                                              AbstractTree{NDIMS}}
-    # TODO: Taal refactor, allow other RealT for the mesh, not just Float64
-    TreeMesh{NDIMS, TreeType}(n_cells_max, SVector{NDIMS, Float64}(domain_center),
-                              convert(Float64, domain_length), periodicity)
+function TreeMesh{NDIMS, TreeType, RealT}(n_cells_max::Integer,
+                                          domain_center::NTuple{NDIMS, RealT},
+                                          domain_length::RealT,
+                                          periodicity = true) where {NDIMS,
+                                                                     TreeType <:
+                                                                     AbstractTree{NDIMS},
+                                                                     RealT <: Real}
+    TreeMesh{NDIMS, TreeType, RealT}(n_cells_max, SVector{NDIMS, RealT}(domain_center),
+                                     domain_length, periodicity)
 end
 
 function TreeMesh(coordinates_min::NTuple{NDIMS, Real},
```

Comment:
Wouldn't it be better if the types of the input arguments `coordinates_min` and `coordinates_max` determined the real type used by the mesh instead of having to pass an additional keyword argument?

Reply:
Yeah I also thought about this in the first place. With this version we avoid a breaking change, though. And it is more similar to the other mesh constructors which have the `RealT` keyword.

Reply:
Sounds reasonable. Although I think it can be confusing if you specify the coordinates as `Float32` and still get a `Float64` mesh.

Reply:
Absolutely, one could/should maybe throw a warning if the supplied coordinates do not match the type.